pints/_optimisers/_adam.py

```diff
@@ -0,0 +1,163 @@
+#
+# Adam optimiser.
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import pints
+
+import numpy as np
+
+
+class Adam(pints.Optimiser):
+    """
+    Adam optimiser (adaptive moment estimation), as described in [1]_.
+
+    This method is a variation on gradient descent that maintains two
+    "moments", allowing it to overshoot and go against the gradient for a short
+    time. This property can make it more robust against noisy gradients. Full
+    pseudo-code is given in [1]_ (Algorithm 1).
+
+    This implementation uses a fixed step size, set as `` min(sigma0)``. Note
+    that the adaptivity in this method comes from the changing moments, not
+    the step size.
+
+    References
+    ----------
+    .. [1] Adam: A method for stochastic optimization
+           Kingma and Ba, 2017, arxiv (version v9)
+           https://doi.org/10.48550/arXiv.1412.6980
+    """
+
+    def __init__(self, x0, sigma0=0.1, boundaries=None):
+        super().__init__(x0, sigma0, boundaries)
+
+        # Set optimiser state
+        self._running = False
+        self._ready_for_tell = False
+
+        # Best solution found
+        self._x_best = self._x0
+        self._f_best = float('inf')
+
+        # Current point, score, and gradient
+        self._current = self._x0
+        self._current_f = np.inf
+        self._current_df = None
+
+        # Proposed next point (read-only, so can be passed to user)
+        self._proposed = self._x0
+        self._proposed.setflags(write=False)
+
+        # Moment vectors
+        self._m = np.zeros(self._x0.shape)
+        self._v = np.zeros(self._x0.shape)
+
+        # Powers of decay rates
+        self._b1t = None
+        self._b2t = None
+
+        # Exponential decay rates for the moment estimates
+        self._b1 = 0.9    # 0 < b1 <= 1
+        self._b2 = 0.999  # 0 < b2 <= 1
+
+        # Step size
+        self._alpha = np.min(self._sigma0)
```

Comment:
Is there a reason that we initialise this method with `sigma0 = 0.1` and not with the recommended alpha = 0.001 from Algorithm 1 in the reference?

Reply:
It's the default value for all optimisers. But I suppose I can change it here!